SerialPrograms/Source/PokemonSV/Programs/Sandwiches/PokemonSV_SandwichRoutines.cpp

```diff
@@ -262,6 +263,42 @@ std::string box_to_string(const ImageFloatBox& box){
     return os.str();
 }
 
+/* 
+- center the cursor by moving the cursor to the edge of the screen, then away from the edge.
+- then search the whole screen for the sandwich hand, instead of just the box, and
+update the location of the sandwich hand
+- return true if successful. else throw an exception
+*/
+bool move_then_recover_sandwich_hand_position(

```

Comment:
Does this work if you're holding down A to hold an ingredient?

Reply:
I'll have to double check later, but I think that it drops the ingredient when it tries the recovery sequence.

Reply:
If we want the hand to keep holding down A, do I need to use `move_session.dispatch([&](BotBaseContext& context)`?

Reply:
Not sure. There's two ways to do this with different levels of elegance and error-proneness.

The elegant but error-prone method is to use `ssf_` and issue a very long A press that returns immediately. Then do all the movement using only `ssf_` touching the joystick buttons only. This effectively "holds" the A button while you do other stuff. But you'll eventually need to issue a cancel to stop the A press.

The less elegant, but safer method is to take whether A is being held down as a parameter. Then on every joystick movement, you either press the joysticks alone, or you use `pbf_controller_state()` to simultaneously press both A and joystick - being careful not to insert any waits that would imply releasing all buttons (including the A).

I *think* Gin's current code is the latter. `ssf_` wasn't available when Gin first write all this. And `ssf` is hard to use if you don't know how it works.